query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala

```diff
@@ -164,6 +171,54 @@ object RangeVectorAggregator {
       }
     }
   }
+
+  /**
+   * A fast reduce method intended specifically for the case when no grouping needs to be done AND
+   * the previous transformer is a PeriodicSampleMapper with fixed output lengths.
+   * It's much faster than mapReduce() since it iterates through each vector first and then from vector to vector.
+   * Time wise first iteration also uses less memory for high-cardinality use cases and reduces the
+   * time window of holding chunk map locks to each time series, instead of the entire query.
+   */
+  def fastReduce(rowAgg: RowAggregator,
+                 skipMapPhase: Boolean,
```

Comment:
It looks like skipMapPhase is always false for now. When would it be true?

Reply:
I believe `skipMapPhase` is true in higher level reduces - ie the phase that takes intermediate aggregates from each leaf/shard and aggregates them further.